resident/resident-service/src/main/java/io/mosip/resident/service/impl/ProxyMasterdataServiceImpl.java

```diff
@@ -413,5 +462,118 @@ public ResponseWrapper<?> getGenderTypesByLangCode(String langCode) throws Resid
 		logger.debug("ProxyMasterdataServiceImpl::getGenderTypesByLangCode()::exit");
 		return responseWrapper;
 	}
+  
+	/**
+	 * download registration centers based on language code, hierarchyLevel and center names
+	 */
+	public InputStream downloadRegistrationCentersByHierarchyLevel(String langCode, Short hierarchyLevel,
+			List<String> name) throws ResidentServiceCheckedException,IOException,Exception {		
+		logger.debug("ResidentServiceImpl::getResidentServicePDF()::entry");
+		ResponseWrapper<?> proxyResponseWrapper = proxyMasterdataService
+				.getAllTemplateBylangCodeAndTemplateTypeCode(langCode, REGISTRATION_CENTER_TEMPLATE_NAME);		
+		ResponseWrapper<?> regCentResponseWrapper = proxyMasterdataService.getRegistrationCentersByHierarchyLevel(langCode,hierarchyLevel, name);
+		Map<String,Object> regCentersMap = new LinkedHashMap<>();
+		List<RegistrationCenterDto >regCentersDtlsList=Collections.emptyList();		
+		if (regCentResponseWrapper != null) {
+			RegistrationCenterResponseDto registrationCentersDtls = mapper.readValue(
+					mapper.writeValueAsString(regCentResponseWrapper.getResponse()), RegistrationCenterResponseDto.class);
+			List<RegistrationCenterDto> regCenterIntialList = registrationCentersDtls.getRegistrationCenters();
+			if(regCenterIntialList!=null && !regCenterIntialList.isEmpty()) {
+				IntStream.range(0, regCenterIntialList.size()).forEach(i-> {
```

Comment:
wow this is a killer. In some of the countries the list is so huge and going over one by one will be a disaster. why do we need to download the entire list? Is there are requirement to download all the center or just a specific hierarchy?

Reply:
just a specific hierarchy. Based on user search criteria